contrib/test_jenkins.sh

```diff
@@ -923,40 +915,60 @@ run_gtest() {
 		GTEST_EXTRA_ARGS="$GTEST_EXTRA_ARGS -p 10 -i 2.0"
 	fi
 	export GTEST_EXTRA_ARGS
+}
 
-	# Run all tests
-	echo "==== Running unit tests, $compiler_name compiler ===="
-	$AFFINITY $TIMEOUT make -C test/gtest test
-
+unset_test_flags() {
 	unset GTEST_EXTRA_ARGS
-
-	# Run valgrind tests
-	if ! [[ $(uname -m) =~ "aarch" ]] && ! [[ $(uname -m) =~ "ppc" ]] && \
-	   ! [[ -n "${JENKINS_NO_VALGRIND}" ]]
-	then
-		echo "==== Running valgrind tests, $compiler_name compiler ===="
-
-		# Load newer valgrind if naative is older than 3.10
-		if ! (echo "valgrind-3.10.0"; valgrind --version) | sort -CV
-		then
-			module load tools/valgrind-3.12.0
-		fi
-
-		$AFFINITY $TIMEOUT_VALGRIND make -C test/gtest test_valgrind
-		module unload tools/valgrind-3.12.0
-	else
-		echo "==== Not running valgrind tests with $compiler_name compiler ===="
-	fi
-
 	unset GTEST_REPORT_LONGEST_TESTS
 	unset GTEST_TOTAL_SHARDS
 	unset GTEST_SHARD_INDEX
+
 	unset OMP_NUM_THREADS
 	unset GTEST_UCT_TCP_FASTEST_DEV
 	unset GTEST_SHUFFLE
 	unset GTEST_RANDOM_SEED
 }
 
+run_specific_tests() {
+	set_common_test_flags
+
+	# Run specific tests
+	do_distributed_task 1 4 run_malloc_hook_gtest
+	do_distributed_task 2 4 run_gtest_watchdog_test 5 60 300
+	do_distributed_task 3 4 test_memtrack
+
+	unset_test_flags
+}
+
+#
+# Run the test suite (gtest)
+# Arguments: <compiler-name> <make-target> [configure-flags]
+#
+run_gtest_make() {
+	compiler_name=$1
+	make_target=$2
+
+	set_common_test_flags
+	set_gtest_test_flags
```

Comment:
`unset GTEST_EXTRA_ARGS` is missing before running valgrind tests.

Reply:
Good catch, fixed!